cloud_testing/xctest_parser/src/main/kotlin/xctest/Shard.kt

```diff
@@ -0,0 +1,81 @@
+package xctest
+
+import com.dd.plist.NSArray
+import com.dd.plist.NSDictionary
+import com.dd.plist.PropertyListParser
+import java.io.File
+import java.nio.file.Paths
+
+object Shard {
+
+    private fun testsForTarget(testDictionary: NSDictionary, testTarget: String, testRoot: String): Set<String> {
+        val productPaths = (testDictionary["DependentProductPaths"] as NSArray)
+        for (product in productPaths.array) {
+            val productString = product.toString()
+            if (productString.contains("/$testTarget.xctest")) {
+                val binaryRoot = productString.replace("__TESTROOT__/", testRoot)
+                println("Found product: $binaryRoot")
+
+                val isSwift = Paths.get(binaryRoot, "libswiftRemoteMirror.dylib").toFile().exists()
+                val binaryName = File(binaryRoot).nameWithoutExtension
+                val binaryPath = Paths.get(binaryRoot, binaryName).toString()
+
+                val methods = if (isSwift) {
+                    println("Swift detected")
+                    Parse.parseSwiftTests(binaryPath)
+                } else {
+                    println("Objc detected")
+                    Parse.parseObjcTests(binaryPath)
+                }
+
+                return methods
+            }
+        }
+
+        return setOf()
```

Comment:
Why would we have a shard without tests? Shouldn't we raise an exception here?

Reply:
Good point, yeah I think we should raise an exception here since we don't want empty shards.